### ISSUE TITLE
webengage ios  push notification click not working. getting exception upon following reference docs.

### ISSUE THREAD

**Issue by chaitanya71998** [WebEngage/react-native-webengage#51]
Hey 👋🏻 
We tried doing integration of push notification call back in ios of our react-native app. but ended up getting this exception in main thread. Why is this happening. 
doc followed: https://docs.webengage.com/docs/react-native-callbacks#push-notification-callbacks
exception:
```
Thread 1: "-[WEGWebEngageBridge isLoading]: unrecognized selector sent to instance 0x2839ef3e0"
```
in file (main.m) at  UIApplicationMain
```
#import <UIKit/UIKit.h>

#import "AppDelegate.h"

int main(int argc, char *argv[])
{
  @autoreleasepool {
    return UIApplicationMain(argc, argv, nil, NSStringFromClass([AppDelegate class]));
  }
}
```

my app delegate file:
```
#import "AppDelegate.h"
#import <WebEngage/WebEngage.h>
#import <React/RCTBundleURLProvider.h>
#import <Firebase.h>
#import <React/RCTRootView.h>


@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
{
  self.bridge = [WEGWebEngageBridge new];
  RCTBridge *bridge = [[RCTBridge alloc] initWithDelegate:self.bridge launchOptions:launchOptions];
  RCTRootView *rootView = [[RCTRootView alloc] initWithBridge:bridge moduleName:@"Bytes" initialProperties:nil];
  self.moduleName = @"Bytes";
  self.initialProps = @{};
  // Enable Firebase Analytics
  [FIRApp configure];
  [FIRAnalytics setAnalyticsCollectionEnabled:YES];
   [WebEngage sharedInstance].pushNotificationDelegate = self.bridge;
  [[WebEngage sharedInstance] application:application didFinishLaunchingWithOptions:launchOptions];
  [[WebEngage sharedInstance] autoTrackUserLocationWithAccuracy:WEGLocationAccuracyDisable];

  return [super application:application didFinishLaunchingWithOptions:launchOptions];
}
- (NSURL *)sourceURLForBridge:(RCTBridge *)bridge
{
#if DEBUG
  return [[RCTBundleURLProvider sharedSettings] jsBundleURLForBundleRoot:@"index"];
#else
  return [[NSBundle mainBundle] URLForResource:@"main" withExtension:@"jsbundle"];
#endif
}
- (BOOL)concurrentRootEnabled
{
  return true;
}
@end

```

in react native implementation:

```
webenage.push.click((notificationData)=>{
console.log('Data')
})
```

How to solve this? is there any fix to get callback triggered without this?

TIA

**Comment by felipegmg** [comment]
same issue here!

**Comment by MilindWebEngage** [comment]
@felipegmg If you are experiencing a crash after sending the push notification, you need to disable the swizzling for it. You can follow the documentation provided in the link [here](https://docs.webengage.com/docs/ios-advanced#7-disable-swizzling) to disable the swizzling.

Additionally, please add the following line before the return statement in `didFinishLaunchingWithOptions` in the app delegate:

```
 if (@available(iOS 10.0, *)) {
    [UNUserNotificationCenter currentNotificationCenter].delegate = (id<UNUserNotificationCenterDelegate>) self;
  }
```

This issue may occur due to having multiple push services in a single app (Firebase and WebEngage). Please also check the documentation for Firebase for further assistance.

If you have any queries, feel free to reach out to us at [support@webengage.com](mailto:support@webengage.com).

**Comment by chaitanya71998** [comment]
Yes that's correct, i was using multiple push services(Firebase and WebEngage), Upon clicking notification For android, notification events, are getting triggered and getting the notification data only if we have `@react-native-firebase/messaging`, but in ios if we have this package it is causing app crash upon clicking notification. 
So I handled this with beautiful solution provided by react native
```
//react-native.config.js
module.exports = {
  dependencies: {
    '@react-native-firebase/messaging': {
      platforms: {
        ios: null,
      },
    },
  },
}
```
